### ISSUE TITLE
Clicking the select file button will freeze on macOS 14.4 arm64

### ISSUE THREAD

**Issue by jiayouzl** [brentvollebregt/auto-py-to-exe#473]
macos 14.4 arm64 最新版本的auto-to-exe无法选择文件,点击选择文件按钮,就卡死了.

**Comment by brentvollebregt** [comment]
When you say "the latest version" - does an older version work for you currently?

**Comment by jiayouzl** [comment]
> When you say "the latest version" - does an older version work for you currently?

All have this problem!

**Comment by brentvollebregt** [comment]
Does this look related to #304 by any chance?

If you run this script in Python, does it lead to the same issue?

```python
from tkinter import Tk
from tkinter.filedialog import askopenfilename

root = Tk()
root.withdraw()
root.wm_attributes('-topmost', 1)
file_path = askopenfilename(parent=root)
root.update()
print(file_path)
```

**Comment by jiayouzl** [comment]
> Does this look related to #304 by any chance?
> 
> If you run this script in Python, does it lead to the same issue?
> 
> ```python
> from tkinter import Tk
> from tkinter.filedialog import askopenfilename
> 
> root = Tk()
> root.withdraw()
> root.wm_attributes('-topmost', 1)
> file_path = askopenfilename(parent=root)
> root.update()
> print(file_path)
> ```

Yes, Python 3.9.19 encountered the same problem when running.

**Comment by jiayouzl** [comment]
```
root.withdraw()
```

Delete the code line above and it will be normal.

**Comment by brentvollebregt** [comment]
Great find, however when removing `root.withdraw()`, I imagine there is another small window that appears which looks like this:

![image](https://github.com/brentvollebregt/auto-py-to-exe/assets/19944838/481b3308-6ee0-489a-a9b7-19a9aefda7a0)

Do you also get this same window?

**Comment by jiayouzl** [comment]
> Great find, however when removing `root.withdraw()`, I imagine there is another small window that appears which looks like this:
> 
> ![image](https://private-user-images.githubusercontent.com/19944838/316290263-481b3308-6ee0-489a-a9b7-19a9aefda7a0.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTEyNjM3MzYsIm5iZiI6MTcxMTI2MzQzNiwicGF0aCI6Ii8xOTk0NDgzOC8zMTYyOTAyNjMtNDgxYjMzMDgtNmVlMC00ODlhLWE5YjctMTlhOWFlZmRhN2EwLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAzMjQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMzI0VDA2NTcxNlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWEwN2Y2NTJiNDRjZjEzYzhkZmNkODBhYTRiMzU4YzJhZTE5Yjc5MWQ4OTY1ODAyYzdiN2Q0NGZjMzFhMWEwZDAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.NMV3Z_rlzatRb9KRryR1KTTPyefYC35Ye6A5EhQ6RK0)
> 
> Do you also get this same window?
> 
> Aside from answering that question, could you also try this:
> 
> ```python
> from tkinter import Tk
> from tkinter.filedialog import askopenfilename
> 
> root = Tk()
> root.withdraw()
> # root.wm_attributes('-topmost', 1)
> file_path = askopenfilename(parent=root)
> root.update()
> print(file_path)
> ```
> 
> Wondering if `root.wm_attributes('-topmost', 1)` is the issue as it's trying to interact with a window that's no longer there.

No, the problem is not solved.

**Comment by jiayouzl** [comment]
I tried to change the code, and the final code is as follows. You can refer to it, and my test here is already working properly.

```
# -*- coding: UTF-8 -*-

from tkinter import Tk
from tkinter.filedialog import askopenfilename

root = Tk()
root.withdraw()

root.wm_attributes('-topmost', 1)
file_path = askopenfilename()
root.update()
print(file_path)
```

**Comment by brentvollebregt** [comment]
Oh, that's really interesting; dropping `parent=root` from the `askopenfilename` call. Thank you very much for testing this - I'll take a look at that parameter and see if it makes sense to drop.

**Comment by brentvollebregt** [comment]
I've just released [auto-py-to-exe 2.43.2](https://github.com/brentvollebregt/auto-py-to-exe/releases/tag/v2.43.2) with this change - hopefully this fixes the issue. Let me know if it doesn't and we can look into this more.

**Comment by jiayouzl** [comment]
> I've just released [auto-py-to-exe 2.43.2](https://github.com/brentvollebregt/auto-py-to-exe/releases/tag/v2.43.2) with this change - hopefully this fixes the issue. Let me know if it doesn't and we can look into this more.

Okay, 2.43.2 can be used now.

There are still some minor issues, such as after selecting a file once, when selecting the file again, the file selection dialog is in the background of the program instead of in the foreground, and you have to manually bring it up.

**Comment by brentvollebregt** [comment]
> There are still some minor issues, such as after selecting a file once, when selecting the file again, the file selection dialog is in the background of the program instead of in the foreground, and you have to manually bring it up.

I've just put a change on the master branch that may fix this. Instead of creating a new `tkinter.Tk` object each time, only one is created.
